### ISSUE TITLE
Cant set up an environment in order to contribute to the development of GlobaLeaks

### ISSUE THREAD

**Issue by connectnet** [globaleaks/GlobaLeaks#3652]
### What version of GlobaLeaks are you using?

[4.13.11]

### What browser(s) are you seeing the problem on?

Other

### What operating system(s) are you seeing the problem on?

Linux

### Describe the issue

I try to install from git with this documentation -> [https://docs.globaleaks.org/en/main/developer/Environment.html](url)

**After user# npm install -d**

```
lodash  <=4.17.20
Severity: critical
Regular Expression Denial of Service (ReDoS) in lodash - https://github.com/advisories/GHSA-x5rq-j2xg-h7qm
Prototype Pollution in lodash - https://github.com/advisories/GHSA-4xc9-xhrj-v574
Prototype Pollution in lodash - https://github.com/advisories/GHSA-fvqr-27wr-82fm
Prototype Pollution in lodash - https://github.com/advisories/GHSA-p6mc-m468-83gw
Command Injection in lodash - https://github.com/advisories/GHSA-35jh-r3h4-6jhm
Regular Expression Denial of Service (ReDoS) in lodash - https://github.com/advisories/GHSA-29mw-wpgm-hmr9
Prototype Pollution in lodash - https://github.com/advisories/GHSA-jf85-cpcp-j695
fix available via `npm audit fix --force`
Will install grunt-usemin@0.1.13, which is a breaking change
node_modules/grunt-usemin/node_modules/lodash
  grunt-usemin  >=2.0.0
  Depends on vulnerable versions of lodash
  node_modules/grunt-usemin

6 vulnerabilities (4 moderate, 1 high, 1 critical)

To address all issues (including breaking changes), run:
  npm audit fix --force
```

**After user# pip3 install -r requirements.txt**

```
DEPRECATION: zope.interface is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for zope.interface ... done
  DEPRECATION: sqlalchemy is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for sqlalchemy ... done
  DEPRECATION: cffi is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for cffi ... error
  error: subprocess-exited-with-error
  
  × Running setup.py install for cffi did not run successfully.
  │ exit code: 1
  ╰─> [40 lines of output]
      /home/user1/GlobaLeaks/backend/env/lib/python3.11/site-packages/setuptools/config/setupcfg.py:515: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
        warnings.warn(msg, warning_class)
      running install
      /home/user1/GlobaLeaks/backend/env/lib/python3.11/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-cpython-311
      creating build/lib.linux-x86_64-cpython-311/cffi
      copying cffi/api.py -> build/lib.linux-x86_64-cpython-311/cffi
      copying cffi/cffi_opcode.py -> build/lib.linux-x86_64-cpython-311/cffi
      copying cffi/error.py -> build/lib.linux-x86_64-cpython-311/cffi
      copying cffi/model.py -> build/lib.linux-x86_64-cpython-311/cffi
      copying cffi/lock.py -> build/lib.linux-x86_64-cpython-311/cffi
      copying cffi/vengine_gen.py -> build/lib.linux-x86_64-cpython-311/cffi
      copying cffi/pkgconfig.py -> build/lib.linux-x86_64-cpython-311/cffi
      copying cffi/backend_ctypes.py -> build/lib.linux-x86_64-cpython-311/cffi
      copying cffi/commontypes.py -> build/lib.linux-x86_64-cpython-311/cffi
      copying cffi/cparser.py -> build/lib.linux-x86_64-cpython-311/cffi
      copying cffi/__init__.py -> build/lib.linux-x86_64-cpython-311/cffi
      copying cffi/ffiplatform.py -> build/lib.linux-x86_64-cpython-311/cffi
      copying cffi/recompiler.py -> build/lib.linux-x86_64-cpython-311/cffi
      copying cffi/setuptools_ext.py -> build/lib.linux-x86_64-cpython-311/cffi
      copying cffi/verifier.py -> build/lib.linux-x86_64-cpython-311/cffi
      copying cffi/vengine_cpy.py -> build/lib.linux-x86_64-cpython-311/cffi
      copying cffi/_cffi_include.h -> build/lib.linux-x86_64-cpython-311/cffi
      copying cffi/parse_c_type.h -> build/lib.linux-x86_64-cpython-311/cffi
      copying cffi/_embedding.h -> build/lib.linux-x86_64-cpython-311/cffi
      copying cffi/_cffi_errors.h -> build/lib.linux-x86_64-cpython-311/cffi
      running build_ext
      building '_cffi_backend' extension
      creating build/temp.linux-x86_64-cpython-311
      creating build/temp.linux-x86_64-cpython-311/c
      x86_64-linux-gnu-gcc -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -DUSE__THREAD -DHAVE_SYNC_SYNCHRONIZE -I/usr/include/ffi -I/usr/include/libffi -I/home/user1/GlobaLeaks/backend/env/include -I/usr/include/python3.11 -c c/_cffi_backend.c -o build/temp.linux-x86_64-cpython-311/c/_cffi_backend.o
      c/_cffi_backend.c:15:10: fatal error: ffi.h: Datei oder Verzeichnis nicht gefunden
         15 | #include <ffi.h>
            |          ^~~~~~~
      compilation terminated.
      error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> cffi

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```

### Proposed solution

Non? I dont know how to fix that?

**Comment by connectnet** [comment]
1. -> I just got some npm warnings
2. -> I took a fresh installation from debian-12.1.0-amd64-netinst.iso
3. pip - V -> pip 23.0.1 from /usr/lib/python3/dist-packages/pip (python 3.11)
4. The error was:
_DEPRECATION: cffi is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559_

So  I installed ciffi with --use-pep517:
**pip3 install cffi --use-pep517**

5. Than I changed the requirements in -> GlobaLeaks/backend/requirements to

```
acme
automat
chardet
cffi
cryptography
h2
idna
priority
pyopenSSL
pyasn1
pyasn1_modules
pynacl
pyotp
python_debian
python_gnupg
requests
service_identity
sqlalchemy
twisted
txtorcon
zope.interface
```
7. Final is error now:
` [stdout#info] [E] Failed to initialize Tor connection; error: Tor control port not open on /home/user1/GlobaLeaks/backend/workingdir/tor/tor_control; waiting for Tor to become available`

Any ideas for fixing to error?
